apps_product_creator/models/odoo_module.py

```diff
@@ -0,0 +1,131 @@
+            []
+        )
+        # choose the view_mode accordingly
+        if len(product_ids) > 1:
+            result['domain'] = "[('id','in',[" + ','.join(
+                map(str, product_ids)) + "])]"
+        elif len(product_ids) == 1:
+            res = self.env.ref('product.product_normal_form_view', False)
+            result['views'] = [(res and res.id or False, 'form')]
+            result['res_id'] = product_ids and product_ids[0] or False
+        return result
+
+    @api.multi
+    def action_create_product(self):
+        self._create_product()
+
+    @api.multi
+    def _create_product(self):
+        """
+        Create the product template related to the module in current recordset.
+        :return: product.template recordset
+        """
+        product_obj = self.env['product.template']
+        products = self.env['product.template']
+        modules = self.filtered(lambda m: not m.product_template_id)
+        domain = [
+            ('odoo_module_id', 'in', modules.ids),
+        ]
+        matching_products = product_obj.search(domain)
+        for odoo_module in modules:
+            product = matching_products.filtered(
+                lambda p: p.odoo_module_id == odoo_module)
+            if not product:
+                product_values = odoo_module._get_template_values()
+                new_product = product_obj.create(product_values)
+                odoo_module.write({
+                    'product_template_id': new_product.id,
+                })
+                products |= new_product
+        return products
+
+    @api.multi
+    def _get_template_values(self):
+        """
+        Create the dict to create a product.template recordset based on the
+        current recordset.
+        The values dict contains info to link the future product with the
+        current module. It also fill the name of the future product with the
+        name of the current module.
+        :return: dict
+        """
+        self.ensure_one()
+        attribute_obj = self.env['product.attribute.value']
+        series = self.module_version_ids.mapped(
+            'repository_branch_id.organization_serie_id.name')
+        attributes = attribute_obj.search([('name', 'in', series)])
+        attribute = self.env.ref(
+            'apps_product_creator.attribute_odoo_version')
+        attribute_line_values = {
+            'attribute_id': attribute.id,
+            'value_ids': [(6, 0, attributes.ids)],
+        }
+        values = {
+            'odoo_module_id': self.id,
+            'type': 'service',
+            'name': self.name,
+            'purchase_ok': False,
+            'list_price': 0,
+            'standard_price': 0,
+            'image': self.image,
+            'attribute_line_ids': [
+                (0, 0, attribute_line_values),
+            ]
+        }
+        return values
+
+    @api.multi
+    def write(self, values):
+        """
```

Comment:
I think that comment on write (and above on create) functions are useless, as it is a standard in Odoo.